### ISSUE TITLE
Incomplete translations show up in search results

### ISSUE THREAD

**Issue by mgeisler** [google/comprehensive-rust#2327]
I've been hesitant to generate and publish HTML for half-done translations since there is a risk that search engines get confused by the duplicate content. A half-done translation will result in lots and lots of pages with English content. We emit a correct `<html lang="xx">` tag, so I guess it should be fine, but it's something to be aware of.

_Originally posted by @mgeisler in https://github.com/google/comprehensive-rust/pull/2189#discussion_r1670615250_

**Comment by mgeisler** [comment]
I just noticed that this is starting to happen. I searched for "async rust comprehensive" because I wanted to find one of our examples, and the top result points to the [Bengali translation](https://google.github.io/comprehensive-rust/bn/async.html):

![image](https://github.com/user-attachments/assets/084207e9-af2a-489f-b6f5-888e28473b7f)

My browser should tell Google that I can read English, Danish, and German, but I still got this result — the `<html lang="bn"` at the top of the page has not been a strong enough signal here.

We could consider adding a no-index tag to very incomplete translations?

Cc @qwandor, @henrif75, @djmitche.

**Comment by henrif75** [comment]
Maybe we could add a `robots.txt` to the directory of unfinished translations to avoid crawling? In that case, we would have to add a parameter to tell which ones are good for publishing.

**Comment by mgeisler** [comment]
> Maybe we could add a `robots.txt` to the directory of unfinished translations to avoid crawling? In that case, we would have to add a parameter to tell which ones are good for publishing.

Yes, we could do something like that! We cannot use `robots.txt` since it has to be in the root of the domain (last I checked), but we can inject the `no-index` attribute in the theme somehow.

**Comment by mgeisler** [comment]
> In that case, we would have to add a parameter to tell which ones are good for publishing.

Yeah, good point: we should probably just follow the same principle as for when we include translations in the language picker.

**Comment by djmitche** [comment]
https://stackoverflow.com/questions/32784322/stopping-index-of-github-pages

 * Can use a user page to put `robots.txt` at the root of the domain. I think that would be https://github.com/google/google?
 * `<meta name="robots" content="noindex">` may do the trick - maybe we just add that to incomplete translations?

**Comment by mgeisler** [comment]
> * `<meta name="robots" content="noindex">` may do the trick - maybe we just add that to incomplete translations?

Yes, I believe this is the correct solution for us!
